src/lib/Microsoft.Health.Extensions.Fhir.R4/Telemetry/Exceptions/FhirServiceExceptionProcessor.cs

```diff
@@ -51,11 +50,13 @@ public static (Exception customException, string errorName) CustomizeException(E
                             errorName = nameof(FhirServiceErrorCode.AuthorizationError);
                             return (new UnauthorizedAccessFhirServiceException(message, exception, helpLink, errorName), errorName);
                         case HttpStatusCode.NotFound:
-                            message = FhirResources.FhirServiceNotFound;
+                            message = DetermineNotFoundErrorMessage(exception);
                             errorName = nameof(FhirServiceErrorCode.ConfigurationError);
                             return (new InvalidFhirServiceException(message, exception, errorName), errorName);
-                        default:
+                        case HttpStatusCode.BadRequest:
                             return (exception, status.ToString());
+                        default:
```

Comment:
Previously, it looks like if the status code didn't match any `case` statement, we'd return the status code as the error name. Now we always return `FhirServiceErrorCode.HttpRequestError`. Can you ask @kyclai or @pallar-ms  if that is desired? I know that they did a lot of work with the exceptions and I don't want to mishandle telemetry.

Reply:
Absolutely, I connected with Carrie offline yesterday about the changes made to the FhirServiceExceptionProcessor and FhirServiceValidator. We're coordinating our changes in one PR

@kyclai or @pallar-ms I'll wait for your feedback on the changes made in the FhirServiceExceptionProcessor before resolving

Reply:
Thanks for bringing this up; please see my other comments.

Reply:
Taking a look on comments. Closing this thread